src/wp-includes/load.php

```diff
@@ -321,11 +321,13 @@ function wp_get_development_mode() {
  *
  * @since 6.3.0
  *
- * @param string $mode Development mode to check for. Either 'core', 'plugin', 'theme', or 'all'.
+ * @param string|null $mode Optional. Development mode to check for. Either 'core', 'plugin', 'theme', or 'all'.
+ *                          If no mode is passed, it will check for any development mode. Defaults to null.
  * @return bool True if the given mode is covered by the current development mode, false otherwise.
  */
-function wp_in_development_mode( $mode ) {
+function wp_is_development_mode( $mode = null ) {
```

Comment:
I waffled a bit on this as well. I landed on `null` to avoid situations where people confused `wp_is_development_mode( 'all' )` with `wp_is_development_mod( 'any' )`, but I'm happy to make the string 'any' the default if nothing is passed, i.e. `wp_is_development_mode()`.

Reply:
I disagree with adding a default value. The reason for introducing development modes was to have a granular mechanism to satisfy _specific_ development requirements, for which `WP_DEBUG` was not sufficient (since it's just yes/no). It should be required to pass a development mode to this function, since otherwise it may easily be misunderstood as "Is my site in development mode? Yes or no?"

As far as I understand, this PR was opened in regards to the renaming, so adding a default value is a different conversation. I would suggest we limit it to why it was initially opened.